### PR TITLE
Drop higher-order A* abstractions for HPF

### PR DESCRIPTION
**Pull request by minkezhang**
Higher-level abstractions are not yet implemented anyways, and preemptively adding the infrastructure to support them is not useful here. Botea 2004 also indicates rapidly diminishing returns on efficiency for L > 1.

### CODE REVIEW

File: lib/hpf/graphastar_test.go [hunk 1]
```diff
@@ -119,49 +118,49 @@ func TestPath(t *testing.T) {
 		{
 			name: "TrivialReachablePath",
 			tm:   trivialOpenMap,
-			g:    buildGraphInput{tileDimension: &rtsspb.Coordinate{X: 1, Y: 1}, level: 1},
-			src:  &rtsspb.AbstractNode{Level: 1, TileCoordinate: &rtsspb.Coordinate{X: 0, Y: 0}},
-			dest: &rtsspb.AbstractNode{Level: 1, TileCoordinate: &rtsspb.Coordinate{X: 0, Y: 0}},
+			g:    buildGraphInput{tileDimension: &rtsspb.Coordinate{X: 1, Y: 1}},
+			src:  &rtsspb.AbstractNode{TileCoordinate: &rtsspb.Coordinate{X: 0, Y: 0}},
+			dest: &rtsspb.AbstractNode{TileCoordinate: &rtsspb.Coordinate{X: 0, Y: 0}},
```
Comment (RyanzpLee) [inline lib/hpf/graphastar_test.go:123]: What is this level: 1 that has been removed. For a*, the map was separated into clusters, is level 1 the higher level overview of the cluster paths?

**Review by RyanzpLee — APPROVED** [review]
I saw some references to level 1 being deleted for the A* abstractions, but for the botea HPF, is it not necessary for different levels of clusters? I assume the hierarchical suggests that there are several layers of clusters, which break down into smaller clusters.

Or is this to be added later

**Comment by minkezhang** [comment]
> I saw some references to level 1 being deleted for the A* abstractions, but for the botea HPF, is it not necessary for different levels of clusters? I assume the hierarchical suggests that there are several layers of clusters, which break down into smaller clusters.
> 
> Or is this to be added later, I assume this is what you mean when you said "preemptively adding the infrastructure is not useful here"

The latter -- IMO this is an example of putting the cart before the horse. I'm paring down some cruft that only serves to complicate the basic implementation.

You are correct that Botea does mention multiple level of clusters; the idea there is, as you've stated, to group these clusters together into even higher-order level `Tile` groupings. The trade-off as mentioned in the paper is that we need to insert the source and destination into all levels of the abstract hierarchy, but as we create higher order (and therefore larger) clusters, then finding a path between a point inside the cluster (WLOG source) will also take a longer time (since we're doing A* over a larger area of the base map). The experimental results seem to state like a 10x speed-up from L0 to L1, but only a 2x speed-up from L1 to L2. That, coupled with the fact for L > 1 we need to add non-trivial logic, means I got lazy and decided to kill that potential feature for now.

File: lib/hpf/graphastar_test.go (hunk 1 above)
Comment (minkezhang) [reply lib/hpf/graphastar_test.go:123]: Deleting the `Level` property because everything is implied to be L1.

**Review by RyanzpLee — APPROVED** [review]

**Pull request merged**